File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        int answer = 0;
        int n = citations.length;
        for (int i = 1; i <= n; i++) {
            if(i <= citations[n-i]) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int ans = 0;
    sort(citations.rbegin(), citations.rend());
    for(int i=0; i<citations.size(); i++) {
        if(i+1 <= citations[i])
            ans++;
    }
    return ans;
}